tasks/grunt-postmark-templates.js

```diff
@@ -0,0 +1,113 @@
+/*
+ * grunt-postmark-templates
+ * push templates to a Postmark server for use with SendTemplatedEmail
+ *
+ * https://github.com/wildbit/grunt-postmark.git
+ */
+
+module.exports = function(grunt) {
+
+  'use strict';
+
+  grunt.registerMultiTask('postmark-templates', 'Create or update Postmark templates', function() {
+
+    var done = this.async();
+    var options = this.options();
+    var template = this.data;
+
+    var serverToken = options.serverToken || grunt.config('secrets.serverToken') || grunt.config('secret.postmark.server_token');
+
+    if (!serverToken) {
+      grunt.fail.warn('Missing required option "serverToken" \n');
+    }
+
+    template.name = template.name || this.target;
+
+    if (!template.name) {
+      grunt.fail.warn('Missing required template property "name" \n');
+    }
+
+    if (!template.subject) {
+      grunt.fail.warn('Missing required template property "subject" \n');
+    }
+
+    var postmark = require('postmark');
+    var client = new postmark.Client(serverToken);
+
+    // read the referenced files, but hold on to the original filenames
+    var expanded = Object.assign({}, template);
+
+    if (expanded.htmlBody) {
+      expanded.htmlBody = grunt.file.read(expanded.htmlBody);
+    }
+    if (expanded.textBody) {
+      expanded.textBody = grunt.file.read(expanded.textBody);
+    }
+
+    if (template.templateId) {
+      client.editTemplate(template.templateId, expanded, function(err, response) {
+        if (err && err.code === 1101) {
+          grunt.log.warn('Template ' + template.templateId + ' not found, so attempting create');
+          delete template.templateId;
+          delete expanded.templateId;
+          client.createTemplate(expanded, function(err, response) {
+            handleResponse(err, response, done, template);
+            grunt.log.writeln('Template ' + template.name + ' created: ' + JSON.stringify(response.TemplateId));
```

Comment:
@hybernaut Can you call this before `handleResponse()` on line 54? Seems that `template.name` doesn't come through in the logging because it's being deleted on `handleResonse()`.